cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +105,300 @@ def unset_default_org(self):
+                Path.mkdir(service_type_dir_path)
+
+    def _migrate_services(self):
+        """Migrate .service files from the global_config_dir and
+        any project local directories."""
+        self._migrate_services_from_dir(self.global_config_dir)
+        for local_proj_dir in self._iter_local_project_dirs():
+            self._migrate_services_from_dir(local_proj_dir)
+
+    def _migrate_services_from_dir(self, dir_path: str) -> None:
+        """Migrate all .service files from the given directory to
+        the appropriate service sub-directory and apply the default
+        alias. This is intended to be run against either the
+        global_config_dir, or a local project directory.
+
+        Default aliases are in the form `service_type__scope`.
+        Scope is either 'project' or 'global' depending
+        on where the .service file is located."""
+        if not dir_path:
+            return  # pragma: no cover
+
+        dir_path = Path(dir_path)
+
+        for item in dir_path.iterdir():
+            if item.suffix == ".service":
+                new_service_filepath = self._get_new_service_filepath(item)
+                if not new_service_filepath:
+                    continue
+                if new_service_filepath.is_file():
+                    self.logger.warning(
+                        f"Skipping migration of {item.name} as a default alias already exists for this service type. "
+                    )
+                    continue
+
+                item.replace(new_service_filepath)
+
+    def _get_new_service_filepath(self, item: Path) -> Path:
+        """Given an old .service filepath, determine the path
+        and filename of the new .service file.
+
+        @returns: the Path to the newfile, or None if the service
+        is of an unrecognized type.
+        """
+        service_type = item.name.replace(".service", "")
+        configured_service_types = self.project_config.config["services"].keys()
+        if service_type not in configured_service_types:
+            self.logger.info(f"Skipping migration of unrecognized service: {item.name}")
+            return None
+
+        scope = "global" if item.parent.name == ".cumulusci" else "project"
+        new_filename = f"{service_type}__{scope}.service"
+        services_sub_dir = Path(f"{self.global_config_dir}/services/{service_type}")
+        return services_sub_dir / new_filename
+
+    def _iter_local_project_dirs(self):
+        """Iterate over all local project dirs in ~/.cumulusci"""
+        for item in Path(self.global_config_dir).iterdir():
+            if item.is_dir() and item.name not in ["logs", "services"]:
+                yield item
+
+    def _raise_service_not_configured(self, name):
+        raise ServiceNotConfigured(  # pragma: no cover
+            f"'{name}' service configuration could not be found. "
+            f"Maybe you need to run: cci service connect {name}"
+        )
```

Comment:
I usually figure it's better to write the simple test for a function like this than to ignore coverage.